src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1368,20 +1368,40 @@ private void VisitStatements(ImmutableArray<IOperation> statements)
         {
             for (int i = 0; i < statements.Length; i++)
             {
-                if (statements[i] is VariableDeclarationGroupOperation declarationGroup
-                    && declarationGroup.DeclarationKind != VariableDeclarationKind.Default)
+                if (TryVisitPossibleUsingDeclaration(statements[i], statements, i))
                 {
-                    // visit the using decl with the following statements
-                    var followingStatements = ImmutableArray.Create(statements, i + 1, statements.Length - i - 1);
-                    VisitUsingVariableDeclarationOperation(declarationGroup, followingStatements);
                     break;
                 }
-                else
-                {
-                    VisitStatement(statements[i]);
-                }
             }
+        }
 
+        /// <summary>
+        /// Visits a node that is possibly a using <see cref="IVariableDeclarationGroupOperation"/>
+        /// </summary>
+        /// <param name="operation">The statement to visit</param>
+        /// <param name="statements">All statements in the block containing this node</param>
+        /// <param name="startIndex">The current statement being visited in <paramref name="statements"/></param>
+        /// <returns>True if this visited a using <see cref="IVariableDeclarationGroupOperation"/> operation</returns>
+        /// <remarks>
+        /// The operation being visited is not necessarily equal to statements[startIndex]. 
+        /// When traversing down a set of labels, we set operation to the label.Operation and recurse, but statements[startIndex] still refers to the original parent label 
+        /// as we haven't actually moved down the original statement list
+        /// </remarks>
+        private bool TryVisitPossibleUsingDeclaration(IOperation operation, ImmutableArray<IOperation> statements, int startIndex)
+        {
+            switch (operation)
+            {
+                case IVariableDeclarationGroupOperation declarationGroup when declarationGroup.DeclarationKind == VariableDeclarationKind.Using || declarationGroup.DeclarationKind == VariableDeclarationKind.AsynchronousUsing:
+                    var followingStatements = ImmutableArray.Create(statements, startIndex + 1, statements.Length - startIndex - 1);
+                    VisitUsingVariableDeclarationOperation(declarationGroup, followingStatements);
+                    return true;
+                case ILabeledOperation labelOperation when labelOperation.Operation is object:
+                    VisitLabel(labelOperation.Label);
```

Comment:
>VisitLabel(labelOperation.Label); [](start = 20, length = 33)

Please put Start/FinishVisitingStatement around this case body, similar how VisitLabeled does it.